exercises/practice/phone-number/.meta/example.ex

```diff
@@ -1,106 +1,80 @@
-defmodule Phone do
-  @moduledoc """
-  Utilities to work with phone numbers.
-  """
-  @invalid "0000000000"
-
+defmodule PhoneNumber do
   @doc """
-  Remove formatting from a phone number.
-
-  Returns "0000000000" if phone number is not valid
-  (10 digits or "1" followed by 10 digits)
-
-  ## Examples
-
-  iex> Phone.number("212-555-0100")
-  "2125550100"
-
-  iex> Phone.number("+1 (212) 555-0100")
-  "2125550100"
-
-  iex> Phone.number("+1 (212) 055-0100")
-  "0000000000"
-
-  iex> Phone.number("(212) 555-0100")
-  "2125550100"
-
-  iex> Phone.number("867.5309")
-  "0000000000"
+  Remove formatting from a phone number if the given number is valid. Return an error otherwise.
   """
-  @spec number(String.t()) :: String.t()
-  def number(raw) do
-    # remove decorations
-    raw
-    |> String.replace(~r/[\s+.()-]/, "")
-    |> valid?
-    |> (fn
-          false -> @invalid
-          num -> String.replace(num, ~r/^1/, "")
-        end).()
+  @spec clean(String.t()) :: {:ok, String.t()} | {:error, String.t()}
+  def clean(raw) do
+    validators = [
+      &validate_digits/1,
+      &validate_length/1,
+      &validate_country_code/1,
+      &validate_area_code/1,
+      &validate_exchange_code/1
+    ]
+
+    Enum.reduce_while(validators, {:ok, remove_separators(raw)}, fn validator, acc ->
+      {:ok, number} = acc
+
+      case validator.(number) do
+        {:ok, number} -> {:cont, {:ok, number}}
+        {:error, error} -> {:halt, {:error, error}}
+      end
+    end)
   end
 
-  defp valid?(num) do
-    Regex.match?(~r/^1?(?:[2-9][0-9]{2}){2}\d{4}$/, num)
-    |> (fn
-          true -> num
-          false -> false
-        end).()
+  defp remove_separators(number) do
+    String.replace(number, ~r/[\s+.()-]/, "")
```

Comment:
```suggestion
    String.replace(number, ~r/[^0-9]/, "")
```
Maybe this would be less prone to breaking if the testcases change in the future?

Reply:
We can't strip all non-digits at this point because we need to check later if, after removing those few symbols, there are any not allowed symbols left. The list of symbols that are allowed, but should be removed (so - this regex), didn't really change.

Reply:
Ah, ok!